main.js

```diff
@@ -61,29 +61,34 @@ document.addEventListener("DOMContentLoaded", function () {
 
     todoList.append(elem);
     updateItemsCount(1);
+
+    // Save tasks to localStorage
+    saveTasks();
   }
 
+  // Update the count of items left
   function updateItemsCount(number) {
     itemsLeft.innerText = +itemsLeft.innerText + number;
   }
 
+  // Remove a to-do item
   function removeTodoItem(elem) {
     elem.remove();
     updateItemsCount(-1);
+    saveTasks();
   }
 
-  todoList.addEventListener("click", (event) => {
-    if (event.target.classList.contains("remove")) {
-      removeTodoItem(event.target.parentElement);
-    }
-  });
+  // Save tasks when any change occurs in the task list
+  todoList.addEventListener("change", saveTasks);
```

Comment:
Right now, you're saving all tasks when something changes in one of them, which is not optimized. Try to save only new changes.